test/e2e/fixtures/e2e_suite.go

```diff
@@ -0,0 +1,160 @@
+package fixtures
+
+import (
+	"bufio"
+	"fmt"
+	alpha1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"os"
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/yaml"
+
+	"github.com/argoproj/argo/cmd/argo/commands"
+	"github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+)
+
+var kubeConfig = os.Getenv("KUBECONFIG")
+
+const namespace = "argo"
+
+func init() {
+	if kubeConfig == "" {
+		kubeConfig = filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	}
+	_ = commands.NewCommand()
+}
+
+type E2ESuite struct {
+	suite.Suite
+	client     v1alpha1.WorkflowInterface
+	kubeClient kubernetes.Interface
+}
+
+func (s *E2ESuite) SetupSuite() {
+	_, err := os.Stat(kubeConfig)
+	if os.IsNotExist(err) {
+		s.T().Skip("Skipping test: " + err.Error())
+	}
+}
+
+func (s *E2ESuite) BeforeTest(_, _ string) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		panic(err)
+	}
+	s.kubeClient, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	s.client = commands.InitWorkflowClient()
+	// delete all workflows
+	log.WithFields(log.Fields{"test": s.T().Name()}).Info("Deleting all existing workflows")
+	err = s.client.DeleteCollection(nil, metav1.ListOptions{})
```

Comment:
It shouldn't delete all workflows in the namespace.  E2ESuite should delete only the e2e test workflows. Can we add a label or name prefix in all e2e workflows and delete only those?

Reply:
The way that it currently works is that it deletes all workflows and workflow pods in the namespace. It's currently hard-coded to the "argo" namespace. 

Options are:

1. Let it as is.
2. Use a single namespace.
3. Use labels.

Reply:
I vote for option 3. So, there is no extra context switch/namespace. But it will not delete other workflows that are submitted other than e2e.